### DISCUSSION TITLE
Migrating to vulcan based libraries (from osgEarth)

### DISCUSSION THREAD

**Discussion by yunusyurtturk** [pelicanmapping/rocky#49] [Q&A / Support] [answered]
Hi,
I'm quite new to OpenSceneGraph and osgEarth. I will be mostly dealing with glTF, 3D Tiles and GIS specific features so osgEarth was a great choice at the beginning. However, as I research more, I find out that OpenSceneGraph and osgEarth kinda started becoming legacy.

Now I see there is VulkanSceneGraph and other 3D mapping tools like:
- Rocky
- [vsgCs](https://github.com/timoore/vsgCs)
- [vsgEarth](https://github.com/timoore/vsgEarth)

We will be working specifically with "glTF", "3D Tiles", and "GIS" related tasks so I need a solution that is able to cover all of them.

So, is rocky okay for the requirements and is it mature enough when compared to osgEarth?

Also, I'm not able to find "3D Tiles" or "glTF" related code in rocky. Does it mean rocky  lacks those features? Can it be combined with vsgCs or vsgEarth?

**Comment by gwaldron** [comment, marked as answer]
OpenSceneGraph is in maintenance mode, which just means the SDK has reached maturity and the author is not planning any major new advancements. Not the same thing as "legacy" ... OSG is still is wide use.

osgEarth is a mature project that we continue to advance and support through commercial funding. osgEarth does support 3D-Tiles/glTF with its Cesium Native NodeKit.

Rocky is a fairly new project that aims to render GIS data using VSG instead of OSG. It has partial research funding and is still in the early stages of development. It's not nearly as mature as osgEarth. Rocky does not support 3D-Tiles yet. If someone funds it, we'll add it :)

vsgCS is a project that integrates VSG with Cesium Native SDK to display 3D-Tiles datasets. vsgEarth demonstrates using the osgEarth SDK to fetch data, and VSG to render it. I will defer to @timoore to comment on the current and future status of these projects :)

**Reply by timoore** [reply]
vsgEarth was a prototype for rendering geospatial data from source, like osgEarth does it. It uses osgEarth's elevation and image layers, but it would take a lot of work to implement the rest of osgEarth's features. I haven't done anything with it in almost two years and am not sure that it would compile with current VSG.

vsgCS does use the Cesium Native library to read 3D Tiles from Cesium ion or other sources. I keep it up-to-date with Cesium Native releases, although it has fallen behind VSG master a bit, particularly with respect to shadows. You can render glTF with vsgCS, using either vsgXchange plus assimp or Cesium Native and vsgCs' translator.

I think it's fair to say that vsgEarth was useful to both rocky and vsgCs for exploring how to do geospatial rendering in VSG. It helps that VSG has the necessary features built in already i.e., double precision transforms and "compilation" threads.
